### PR TITLE
Add history feature

### LINKED ISSUE #1845: CLI: No editing the command with arrow keys also doesn't work so it's a lot of retyping/copy-pasting
## Description

Briefly describe the feature that you are requesting.

## Motivation

Explain why this feature is needed.

## Requirements

Write a list of what you want this feature to do.

1. 
2. 
3. 

## Open questions (optional)

Use this section to ask any questions that are related to the feature.

## Are you planning to do it yourself in a pull request?

Yes/No.

### LINKED ISSUE #1851: CLI: Change the Account command interface
## Description

I'm not sure if we should keep the current account command interface.
It doesn't make use of some nice cli stuff like the commands showing up in your terminal history.

I guess we could also make use of [dialoguer](https://github.com/mitsuhiko/dialoguer) (or another lib like [tui-rs](https://github.com/fdehau/tui-rs) or [termion](https://gitlab.redox-os.org/redox-os/termion) ) to create input prompts so users don't need to type everything every time and create a nice tub 🤔

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add history feature
Original PR from @hauju https://github.com/iotaledger/cli-wallet/pull/158

## Links to any relevant issues

Fixes https://github.com/iotaledger/wallet.rs/issues/1845
Fixes https://github.com/iotaledger/wallet.rs/issues/1851

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running commands and modifying them on Windows and Linux

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: cli/src/account.rs [hunk 1]
```diff
@@ -36,13 +38,18 @@ pub async fn account_prompt(account_handle: AccountHandle) -> Result<(), Error>
...
     let alias = {
         let account = account_handle.read().await;
         account.alias().clone()
     };
-    let command: String = Input::new().with_prompt(format!("Account \"{alias}\"")).interact()?;
-
+    let command: String = Input::new()
+        .with_prompt(format!("Account \"{}\"", alias))
+        .history_with(history)
+        .interact_text()?;
```
Comment (thibault-martinez) [inline cli/src/account.rs:52]: So what was the consensus on this then?

Comment (Thoralf-M) [reply cli/src/account.rs:52]: I thought it caused problems on windows, but I tried a few commands and all worked fine

File: cli/src/account_history.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+}
+
+impl<T: ToString> History<T> for AccountHistory {
+    fn read(&self, pos: usize) -> Option<String> {
+        self.history.get(pos).cloned()
+    }
+
+    fn write(&mut self, val: &T) {
+        if self.history.contains(&val.to_string()) {
```
Comment (thibault-martinez) [inline cli/src/account_history.rs]: Save the to_string to avoid doing it again L34

Comment (Thoralf-M) [reply cli/src/account_history.rs]: Doesn't work?

**Review by kwek20 — DISMISSED** [review]

File: cli/src/account_history.rs (hunk 2 above)
Comment (thibault-martinez) [reply cli/src/account_history.rs]: Wdym? Do `let entry = val.to_string();` and then use it ?

Comment (Thoralf-M) [reply cli/src/account_history.rs]: Ah yeah this works, misunderstood you

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**